File: src/main/java/pl/mroz/buddiesapi/infrastructure/container/AccountConfig.java
package pl.mroz.buddiesapi.infrastructure.container;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import pl.mroz.buddiesapi.domain.account.AccountDomainService;
import pl.mroz.buddiesapi.domain.account.AccountDomainServiceFactory;
import pl.mroz.buddiesapi.domain.account.AccountRepository;

@Configuration
public class AccountConfig {

    @Bean
    AccountDomainService accountDomainService(AccountRepository repository) {
        return AccountDomainServiceFactory.accountDomainService(repository);
    }
}

File: src/main/kotlin/buddiesapi/infrastructure/container/AccountConfig.kt
package buddiesapi.infrastructure.container

import buddiesapi.domain.account.AccountDomainService
import buddiesapi.domain.account.AccountDomainServiceFactory
import buddiesapi.domain.account.AccountRepository
import org.springframework.context.annotation.Bean

class AccountConfig {
    @Bean
    fun accountDomainService(repository: AccountRepository): AccountDomainService {
        return AccountDomainServiceFactory.accountDomainService(repository)
    }
}
